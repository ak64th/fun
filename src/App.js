import React from 'react';
import {hot} from 'react-hot-loader';
import styled, {injectGlobal, keyframes} from 'styled-components';
import logo from './logo.svg';

injectGlobal`
  body {
    margin: 0;
    padding: 0;
    font-family: sans-serif;
  }
`;

const Wrapper = styled.div`
  text-align: center;
`;

const Header = styled.div`
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
`;

const Spin = keyframes`
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
`;

const Logo = styled.img`
  animation: ${Spin} infinite 20s linear;
  height: 80px;
`;

const Title = styled.h1`
  font-size: 1.5em;
`;

const Intro = styled.p`
  font-size: large;
`;

const Link = styled.a`
  color: #61dafb;
`;

const Code = styled.code`
  font-weight: 600;
  line-height: 1.25;
`;


const App = () => {
    return (
        <Wrapper>
            <Header>
                <Logo src={logo} alt="logo"/>
                <Title>Welcome to React</Title>
            </Header>
            <Intro>
                To get started, edit <Code>src/App.js</Code> and save to reload.
            </Intro>
            <Link
                href="https://reactjs.org"
                target="_blank"
                rel="noopener noreferrer"
            >
                Learn React
            </Link>
        </Wrapper>
    );
};

export default hot(module)(App);
