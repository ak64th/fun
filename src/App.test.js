import React from 'react';
import {mount} from 'enzyme';
import App, {Header} from './App';

describe('App', () => {
  it('renders correctly', () => {
    const tree = mount(<App/>);
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('text-align', 'center');
  });

  it('renders header with correct styles', () => {
    const tree = mount(<App/>);
    const header = tree.find('div').at(1);
    expect(header).toHaveLength(1);
    expect(header).toHaveStyleRule('background-color', '#222');
    expect(header).toHaveStyleRule('height', '150px');
    expect(header).toHaveStyleRule('padding', '20px');
    expect(header).toHaveStyleRule('color', 'white');
  });
});
