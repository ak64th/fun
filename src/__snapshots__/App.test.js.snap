// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
.c0 {
  text-align: center;
}

.c1 {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.c2 {
  -webkit-animation: iVXCSc infinite 20s linear;
  animation: iVXCSc infinite 20s linear;
  height: 80px;
}

.c3 {
  font-size: 1.5em;
}

.c4 {
  font-size: large;
}

.c6 {
  color: #61dafb;
}

.c5 {
  font-weight: 600;
  line-height: 1.25;
}

<App>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <styled.img
            alt="logo"
            src="logo.svg"
          >
            <img
              alt="logo"
              className="c2"
              src="logo.svg"
            />
          </styled.img>
          <styled.h1>
            <h1
              className="c3"
            >
              Welcome to React
            </h1>
          </styled.h1>
        </div>
      </styled.div>
      <styled.p>
        <p
          className="c4"
        >
          To get started, edit 
          <styled.code>
            <code
              className="c5"
            >
              src/App.js
            </code>
          </styled.code>
           and save to reload.
        </p>
      </styled.p>
      <styled.a
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        <a
          className="c6"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
      </styled.a>
    </div>
  </styled.div>
</App>
`;
